test/test_dataloader.py

```diff
@@ -272,22 +366,30 @@ def test_multiple_dataloaders(self):
         next(loader2_it)
 
     def test_segfault(self):
-        p = multiprocessing.Process(target=_test_segfault)
+        p = ErrorTrackingProcess(target=_test_segfault)
         p.start()
-        p.join(JOIN_TIMEOUT)
+        p.join(3.0 + JOIN_TIMEOUT)
```

Comment:
Why do we need 3+ now?

Reply:
Previously we didn't check the error message. Actually the test failed for re-initializing cuda context at `torch.cuda.current_device()` (from the per-device pin_memory PR). So they failed before they reach the segfault code. After fixing that, I observed that sometimes it needs a slightly longer time.

Reply:
Can we just increase the `JOIN_TIMEOUT` then?